backend/api/src/api/v1/gql/subscription/chat.rs

```diff
@@ -0,0 +1,70 @@
+use async_graphql::{Context, Subscription};
+use async_stream::stream;
+use chrono::{TimeZone, Utc};
+use futures_util::Stream;
+use prost::Message;
+use uuid::Uuid;
+
+use crate::{
+    api::v1::gql::{
+        error::{GqlError, Result, ResultExt},
+        ext::ContextExt,
+        models::chat_message::{ChatMessage, MessageType},
+    },
+    pb,
+};
+
+#[derive(Default)]
+pub struct ChatSubscription;
+
+#[Subscription]
+impl ChatSubscription {
+    // Listen to new messages in chat.
+    pub async fn chat_messages<'ctx>(
+        &self,
+        ctx: &'ctx Context<'_>,
+        #[graphql(desc = "Chat to subscribe to.")] channel_id: Uuid,
+    ) -> Result<impl Stream<Item = Result<ChatMessage>> + 'ctx> {
+        let global = ctx.get_global();
+
+        let welcome_message = ChatMessage {
+            id: Uuid::nil(),
+            author_id: Uuid::nil(),
+            channel_id,
+            content: "Welcome to the chat!".to_string(),
+            created_at: chrono::Utc::now().into(),
+            r#type: MessageType::Welcome,
+        };
```

Comment:
Do we really need to send a welcome message?
Can't we just show the message client-side as soon as the user is connected to the chat?

Reply:
Oh right, didn't think about that